src/Features/Core/Portable/ExtractMethod/OperationStatus.cs

```diff
@@ -4,64 +4,53 @@
 
 #nullable disable
 
-using System.Collections.Generic;
 using System.Collections.Immutable;
-using System.Linq;
 using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.ExtractMethod
+namespace Microsoft.CodeAnalysis.ExtractMethod;
+
+internal partial class OperationStatus
 {
-    internal partial class OperationStatus
+    public OperationStatus(OperationStatusFlag flag, string reason)
     {
-        public OperationStatus(OperationStatusFlag flag, string reason)
-        {
-            Contract.ThrowIfTrue(flag.Succeeded() && flag.HasBestEffort());
-
-            Flag = flag;
-            Reasons = reason == null ? ImmutableArray<string>.Empty : ImmutableArray.Create(reason);
-        }
-
-        private OperationStatus(OperationStatusFlag flag, ImmutableArray<string> reasons)
-        {
-            Contract.ThrowIfTrue(reasons.IsDefault);
-            Contract.ThrowIfTrue(flag.Succeeded() && flag.HasBestEffort());
+        Flag = flag;
+        Reasons = reason == null ? ImmutableArray<string>.Empty : ImmutableArray.Create(reason);
+    }
 
-            Flag = flag;
-            Reasons = reasons;
-        }
+    private OperationStatus(OperationStatusFlag flag, ImmutableArray<string> reasons)
+    {
+        Contract.ThrowIfTrue(reasons.IsDefault);
 
-        public OperationStatus With(OperationStatusFlag flag, string reason)
-        {
-            var newFlag = Flag | flag;
+        Flag = flag;
+        Reasons = reasons;
+    }
 
-            newFlag = (this.Failed() || flag.Failed()) ? newFlag.RemoveFlag(OperationStatusFlag.Succeeded) : newFlag;
-            newFlag = newFlag.Succeeded() ? newFlag.RemoveFlag(OperationStatusFlag.BestEffort) : newFlag;
```

Comment:
>newFlag [](http://example.com/codeflow?start=12&length=7)

Looks like OoperationStatusFlag is now essentially a bool. Any consideration to just changing it as such to simplify this code?

Reply:
yup!  i was having the same thought.  can def simplify this.

Reply:
moved to mop up PR.